src/main/java/org/folio/processing/mapping/mapper/reader/record/MarcRecordReader.java

```diff
@@ -168,12 +176,43 @@ private Value readRepeatableField(MappingRule ruleExpression) {
   }
 
   private String extractValueFromMarcRecord(VariableField field, String marcPath) {
+    String value = EMPTY;
     if (field instanceof DataFieldImpl) {
-      return ((DataFieldImpl) field).getSubfieldsAsString(marcPath.substring(marcPath.length() - 1));
+      value = ((DataFieldImpl) field).getSubfieldsAsString(marcPath.substring(marcPath.length() - 1));
+      return tryFormattingToIsoDate(value);
     } else if (field instanceof ControlFieldImpl) {
-      return ((ControlFieldImpl) field).getData();
+      value = ((ControlFieldImpl) field).getData();
+      return tryFormattingToIsoDate(value);
+    }
+    return value;
+  }
+
+  private String tryFormattingToIsoDate(String stringToFormat) {
```

Comment:
I think it will be better to rename this method like "formatToISODate"

Reply:
but if `stringToFormat` can not be formatted to ISO date format, `stringToFormat` is returned as-is